src/update_agent/actor.rs

```diff
@@ -16,6 +16,26 @@ lazy_static::lazy_static! {
         "zincati_update_agent_last_refresh_timestamp",
         "UTC timestamp of update-agent last refresh tick."
     )).unwrap();
+    static ref FINALIZATION_ATTEMPTS: IntCounter = register_int_counter!(opts!(
+        "zincati_update_agent_finalization_attempts",
+        "Total number of attempts to finalize a staged deployment by the update agent."
+    )).unwrap();
+    static ref FINALIZATION_ALLOWED_BY_STRATEGY: IntCounter = register_int_counter!(opts!(
```

Comment:
I think this allowed/denied can be folded into a single metrics with a `outcome=allowed|denied` label and moved into the `strategy` module. I'm not sure whether to fold the error outcome into this label as well.

Reply:
By "folding", you mean using e.g. `IntCounterVec` right?

Reply:
Yes. There are some examples in this project in the `fleet_lock` module.